scripts/kubestellar-snapshot.sh

```diff
@@ -0,0 +1,579 @@
+#!/usr/bin/env bash
+# Copyright 2024 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Exit on error
+set -e
+
+
+# Global variables
+TIMESTAMP="$(date +%F_%T)"
+TMPFOLDER="$(mktemp -d --tmpdir "kubestellar-snapshot-XXXX")"
```

Comment:
Was the omission of `-p .` here deliberate? Without `-p .`, the temp dir gets made in `/tmp`, which is probably less secure than the user's current working directory.

Reply:
I made a choice to use tmp folder but if you rather use the cwd, I can change to it.